Integrate existing executor tests with the new testing framework

NingLin-P opened the pull request:
This PR integrates all the remaining executor tests with the new test framework.

The only ignored executor test is `fraud_proof_verification_in_tx_pool_should_work`, to enable it we need to support fetching the `call_data` of the fraud proof verifier from the original primary block.

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

NingLin-P committed:
Replace is_bundle_present_in_tx_pool with get_bundle_from_tx_pool

get_bundle_from_tx_pool is used in incoming test and is_bundle_present_in_tx_pool
can be implement by get_bundle_from_tx_pool(..).is_some()

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Add produce_block_with_extrinsics to MockPrimaryNode

produce_block_with_extrinsics is useful when we want to control the exact extrinsic
included in block or for extrinsic that may fail on verification but we still want
to include it in block.

This commit also extract the new block logging in a separated function
so both produce_block_with_slot and produce_block_with_extrinsics can reuse it.

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Integrate test execution_proof_creation_and_verification_should_work with new testing framework

This test is broken and fixed by adding primary_block_info digest to block header

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Integrate test fraud_proof_verification_in_tx_pool_should_work with the new testing framework

The test is still ignored due to we don't support fetch the call_data from the primary chain,
and it will be a ready-made test when we support so, there is also a TODO left to test the
initialize_block fraud proof of block 1 when we can properly handle the genesis receipt on
verify_pre_state_root.

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Integrate test extract_core_domain_wasm_bundle_in_system_domain_runtime_should_work with the new testing framework

Signed-off-by: linning <linningde25@gmail.com>

liuchengxu commented on domains/client/domain-executor/src/tests.rs:
```diff
@@ -256,55 +282,48 @@ async fn set_new_code_should_work() {
+    // fail due to `InvalidTransaction::ExhaustsResources` error thus use `set_heap_pages` instead
+    let set_heap_pages_tx = subspace_test_service::construct_extrinsic(
+        ferdie.client.as_ref(),
+        pallet_sudo::Call::sudo {
```
I believe you can use `pallet_sudo::Call::sudo_unchecked_weight` to avoid the exhausted resources.

NingLin-P replied:
We are still unable to use `set_code` due to it requiring a well-format runtime wasm blob
https://github.com/paritytech/substrate/blob/dd45632f298db02f6946c906296bbf86a74f4f46/frame/system/src/lib.rs#L403
I have updated the comment instead.

liuchengxu replied:
There is an API that skips the runtime code check right below https://github.com/paritytech/substrate/blob/dd45632f298db02f6946c906296bbf86a74f4f46/frame/system/src/lib.rs#L414.

However, in order to extract the system domain code from the new primary runtime, we still need it to be a _proper primary runtime code_ with an arbitrary system domain runtime code included. I'm not sure the requirement of the minimal proper primary runtime code, but anyway, you don't have to use `set_heap_pages` here.

NingLin-P replied:
Seems we do need a proper primary runtime code even for this test, otherwise, the executor will shut down and the test will fail due to:
```
2023-04-04 17:03:03 [//Alice (SystemDomain)] Cannot create a runtime error=Other("cannot deserialize module: InvalidMagic")
2023-04-04 17:03:03 [//Alice (SystemDomain)] Failed to process primary block error=Application(VersionInvalid("cannot deserialize module: InvalidMagic"))
```

liuchengxu replied:
Hmm, I have some thoughts on creating a minimal primary runtime code, but unsure if that works actually. Feel free to revert this test or mark it as incomplete because this PR does not actually cover the new domain runtime are updated accordingly upon the new primary runtime so that we can push this PR forward.

NingLin-P replied:
Okay, I added a TODO for `set_new_code_should_work`

liuchengxu replied:
Think about it again, it's probably better to revert it for clarity and keep it ignored IMO as now it's harder to tell the purpose of this test, and doesn't test what we really want. Before, it was obvious that we would like to have a new primary runtime upgrade containing a new system domain runtime, and then the domain runtime is upgraded accordingly (`assert_eq!(runtime_code, new_runtime_wasm_blob)` means the latest domain runtime code must be updated to the one expected.). Now the domain runtime code remains the same in fact as we don't update it actually. In short, simply update the API `run_mock_primary_node` and keep it ignored, and then we can move forward to the other tests and merge this PR.

I also realize that we should improve the logic of checking whether a primary runtime is updated in case `RuntimeEnvironmentUpdated` will be associated with more events, we should check the corresponding event `CodeUpdated` directly instead.

NingLin-P replied:
Okay, I'm good with both paths.

NingLin-P replied:
The test has been reverted, plz take a look at the last commit.

liuchengxu commented on domains/client/domain-executor/src/tests.rs:
```diff
@@ -386,140 +398,112 @@ async fn pallet_domains_unsigned_extrinsics_should_work() {
+                    authority_stake_weight: Default::default(),
+                    authority_witness: Default::default(),
+                    proof_of_election: ProofOfElection::dummy(DomainId::SYSTEM, pair.public()),
+                }, // TODO: mock ProofOfElection properly
```
Why not adjust the API `produce_slot_and_wait_for_bundle_submission` to return the produced bundle as well so that it don't have constructe it manually and have a proper bundle solution?

NingLin-P replied:
The test `pallet_domains_unsigned_extrinsics_should_work` has been rewritten according to the comment and added more cases, the changes have been put to the original commit, PTAL.

NingLin-P committed:
Integrate test pallet_domains_unsigned_extrinsics_should_work with the new testing framework

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Apply review suggestions

Signed-off-by: linning <linningde25@gmail.com>

vedhavyas reviewed (dismissed):
make sense!
Great work!

NingLin-P committed:
Integrate test set_new_code_should_work with the new testing framework

The test is still ignored due to we can not construct a proper primary runtime code.

Signed-off-by: linning <linningde25@gmail.com>

NingLin-P committed:
Merge branch 'main' into executor-tests
